pkg/payload/task.go

```diff
@@ -112,12 +143,28 @@ func (st *Task) Run(ctx context.Context, version string, builder ResourceBuilder
 	}
 }
 
+// UpdateEffectType defines the effect an update error has on the overall update state.
+type UpdateEffectType string
+
+const (
+	// "None" defines an error as having no affect on the update state.
```

Comment:
nit: We should add Go linter.  I expect it would want these godocs to start with the exposed name, so:

```go
// UpdateEffectNone defines an error as having no affect on the update state.
```

and similar for the other constants below.